android/app/build.gradle

```diff
@@ -17,8 +17,10 @@ android {
         applicationId "io.github.ja2stracciatella"
         minSdkVersion 19
         targetSdkVersion 26
-        versionCode 18
-        versionName "0.18.0"
+
+        File versionFile = file('../../version')
+        versionCode 17
```

Comment:
Shouldn't `versionCode` be just an ever incrementing number? Could we use the CI build number for this (something like `GITHUB_RUN_NUMBER`)?

Reply:
Kind of, yes. From Android docs:

> A positive integer used as an internal version number. This number is used only to determine whether one version is more recent than another, with higher numbers indicating more recent versions. This is not the version number shown to users; that number is set by the versionName setting, below. The Android system uses the versionCode value to protect against downgrades by preventing users from installing an APK with a lower versionCode than the version currently installed on their device.

> The value is a positive integer so that other apps can programmatically evaluate it, for example to check an upgrade or downgrade relationship. You can set the value to any positive integer you want, however you should make sure that each successive release of your app uses a greater value. You cannot upload an APK to the Play Store with a versionCode you have already used for a previous version.

I think using `GITHUB_RUN_NUMBER` would not suit this very well, e.g. if I want to install an older nightly version of the same minor release I'd have the problem that its version number is smaller than the one installed, so I'd have to remove it first. I think its better to just increase it for each release.

What I can think of is to just build it from the version as follows:
- Pad each version (MAJOR, MINOR, PATCH) with some Zeros (lets say 4) and concatenate them: `000000170001`
- Parse that into a number: `170001`

That would return e.g.:
- `0.17.1` -> `170001`
- `0.17.2` -> `170002`
- `0.17.1000` -> `171000`
- `0.18.0` -> `180000`
- `1.0.0` -> `100000000`
- `1.1.1` -> `00010001`

So it would be inceasing for each release.

Reply:
NB:

> Warning: The greatest value Google Play allows for versionCode is 2100000000. 

So the example above would work til version 21.

Reply:
Added this logic, but with padding to 3 characters, so max semver version is `999.999.999` and version code `999999999`.

Reply:
LTGM!